### PR TITLE
crimson/os/seastore: convert omap and collection_manager to interruptible future

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by athanatos** [comment]
Hmm, looks like another bionic gcc hang.

**Comment by athanatos** [comment]
Hmm, the jenkins worker seems to be  Agent 172.21.2.15+braggi15 on trusty?  Is that right?

**Comment by athanatos** [comment]
Hang is reproducible on bionic as well, it seems.  Trying to work around.

**Comment by athanatos** [comment]
That seems to fix it on bionic, let's see if make check works now.

**Comment by tchaikov** [comment]
> Hmm, the jenkins worker seems to be Agent 172.21.2.15+braggi15 on trusty? Is that right?

we've upgraded all jenkins worker running "make check" nodes to focal. if that's the case, should fix it.

File: src/test/crimson/seastore/test_collection_manager.cc [hunk 1]
```diff
@@ -23,6 +23,21 @@ namespace {
...
+#define TEST_COLL_FORWARD(METHOD)					\
+  template <typename... Args>				\
+  auto METHOD(coll_root_t &root, Transaction &t, Args&&... args) const { \
+    return with_trans_intr(						\
+    t,									\
+    [this](auto &t, auto &root, auto&&... args) {			\
+      return collection_manager->METHOD(				\
+        root,								\
+        t,								\
+        std::forward<decltype(args)>(args)...);				\
```
Comment (tchaikov) [inline src/test/crimson/seastore/test_collection_manager.cc:35]: nit, could put
```c++
std::forward<Args>(args)...);
```

**Review by tchaikov — APPROVED** [review]

**Pull request merged**